subprojects/core/src/main/java/org/gradle/execution/plan/DefaultNodeValidator.java

```diff
@@ -29,12 +29,16 @@ public boolean hasValidationProblems(Node node) {
             TypeValidationContext context = new TypeValidationContext() {
                 @Override
                 public void visitTypeProblem(Severity severity, Class<?> type, String message) {
-                    foundProblem.set(true);
+                    if (severity != Severity.CACHEABILITY_WARNING) {
+                        foundProblem.set(true);
+                    }
```

Comment:
I think it would be good to add some comments here about why we skip cacheability warnings.

Reply:
I added a comment.